test/test_sklearn_compat.py

```diff
@@ -359,20 +351,17 @@ def make_random_state(estimator, in_pipeline):
       return rs
 
     estimator = clone(estimator)
-    estimator.set_params(preprocessor=preprocessor)
+    estimator.set_params(preprocessor=preprocessor,
+                         **make_random_state(estimator, False))
     pipeline = make_pipeline(estimator)
-    estimator.fit(*remove_y_quadruplets(estimator, input_data, y),
-                  **make_random_state(estimator, False))
-    pipeline.fit(*remove_y_quadruplets(estimator, input_data, y),
-                 **make_random_state(estimator, True))
+    estimator.fit(input_data, y)
```

Comment:
yeah, I was too when dealing with this at the beginning. It looks like the handling of LSML, which is a quadruplet learner, evolved as the tests for it were added. 

Here the quadruplets and triplets are excluded with the `if` on line 338 as they don't have a `y`. Anyways, I think this could be handled more elegantly by getting a list of the allowed estimators for this.

Reply:
OK, fair enough! It is actually explained in a comment on line 335: it would be nice to update it to mention also triplets

Reply:
Yes, for sure! I thought I already did!